### PR TITLE
Fixing issues with links with empty sources

### PR DESCRIPTION
**Pull request by zivnevo**
Signed-off-by: Ziv Nevo <nevo@il.ibm.com>

### CODE REVIEW

File: pkg/controller/synth_netpols.go [hunk 1]
```diff
@@ -57,14 +57,19 @@ func determineConnectivityPerDeployment(connections []common.Connections) []*Dep
...
-		} else if conn.Link.Resource.Type != "LoadBalancer" && conn.Link.Resource.Type != "NodePort" {
-			netpolPeer := network.NetworkPolicyPeer{PodSelector: getDeployConnSelector(srcDeploy)}
-			ingressNetpolPeer = append(ingressNetpolPeer, netpolPeer)
+
+		if conn.Link.Resource.Type == "LoadBalancer" || conn.Link.Resource.Type == "NodePort" {
+			dstDeploy.addIngressRule([]network.NetworkPolicyPeer{}, target_ports) // in these cases we want to allow traffic from all sources
+		} else {
+			var ingressNetpolPeer []network.NetworkPolicyPeer
+			if len(conn.Source.Resource.Name) == 0 {
+				ingressNetpolPeer = append(ingressNetpolPeer, network.NetworkPolicyPeer{PodSelector: nil}) // blocks all traffic
```
Comment (adisos) [inline pkg/controller/synth_netpols.go:66]: It seems we still have this issue (e.g., https://github.com/np-guard/cluster-topology-analyzer/blob/1caaa090ebb6b6943c6513a4e40e3601b99a0ee4/tests/sockshop/expected_netpol_output.json#L170-L201 ), I think we should avoid adding ingress rules if we want to block all ingress traffic on this case.

**Review by adisos — COMMENTED** [review]
Hi @zivnevo , added some comment.

File: pkg/controller/synth_netpols.go (hunk 1 above)
Comment (zivnevo) [reply pkg/controller/synth_netpols.go]: Good catch! Apparently setting the LabelSelector to `nil` didn't work as expected.

Comment (zivnevo) [reply pkg/controller/synth_netpols.go]: Maybe we need 2 modes for synthesis: "Tight" and "Relaxed".  Tight will work like it does now. Relax will open "source-less" microservices for ingress using their declared ports. What do you think?

Comment (adisos) [reply pkg/controller/synth_netpols.go]: Yes, I think it's a good idea.

**Review by adisos — APPROVED** [review]

**Pull request merged**